src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -814,9 +979,110 @@ export class ShadowGenerator implements IShadowGenerator {
         this._applyFilterValues();
     }
 
+    private _initCascades(): void {
+        let camera = this._scene.activeCamera;
+        if (!camera) {
+            return;
+        }
+
+        // inits and sets all static params related to CSM
+        let engine = this._scene.getEngine();
+        this._viewSpaceFrustums = [];
+        this._viewSpaceBoundingSpheres = [];
+        let nearx = 0;
+        let neary = 0;
+        let farx = 0;
+        let fary = 0;
+
+        // get all internal camera cascaded frustum points
+        let breaks = this._frustumSplit(this.cascades, camera.minZ, this._csmFrustumLength, 0.5);
+        if (camera.fovMode === 0) {
+            nearx = camera.minZ * Math.tan(camera.fov / 2) * engine.getAspectRatio(camera);
+            neary = camera.minZ * Math.tan(camera.fov / 2);
+            farx = this._csmFrustumLength * Math.tan(camera.fov / 2) * engine.getAspectRatio(camera);
+            fary = this._csmFrustumLength * Math.tan(camera.fov / 2);
+        } else if (camera.fovMode === 1) {
+            nearx = camera.minZ * Math.tan(camera.fov / 2);
+            neary = camera.minZ * Math.tan(camera.fov / 2) * engine.getAspectRatio(camera);
+            farx = this._csmFrustumLength * Math.tan(camera.fov / 2);
+            fary = this._csmFrustumLength * Math.tan(camera.fov / 2) * engine.getAspectRatio(camera);
+        }
+
+        // populate the viewSpaceFrustums array
+        for (let i = 0; i < this.cascades + 1; i++) {
+            let temp = [];

```

Comment:
why not pushing all the way directly to the array to prevent the temp ???

Reply:
Nice catch! Fixed!